src/fides/api/ops/models/application_config.py

```diff
@@ -82,3 +112,86 @@ def get_api_set_config(cls, db: Session) -> Dict[str, Any]:
         if config_record:
             return config_record.api_set
         return {}
+
+    @classmethod
+    def get_config_set_config(cls, db: Session) -> Dict[str, Any]:
+        """
+        Utility method to get the config_set config dict
+
+        An empty `dict` will be returned if no config settings have been set through config
+        """
+        config_record = db.query(cls).first()
+        if config_record:
+            return config_record.config_set
+        return {}
+
+    @classmethod
+    def set_api_set_config(
+        cls, db: Session, api_set_dict: Dict[str, Any]
+    ) -> ApplicationConfig:
+        """
+        Utility method to set the `api_set` column on the `applicationconfig`
+        db record with the provided dictionary of data.
+
+        Updates are *merged* with any existing `api_set` data in the db.
+        """
+        return cls.create_or_update(db, data={"api_set": api_set_dict})
+
+    @classmethod
+    def set_config_set_config(
+        cls, db: Session, config: FidesConfig
+    ) -> ApplicationConfig:
+        """
+        Utility method to set the `config_set` column on the `applicationconfig`
+        db record by serializing the given `FidesConfig` to a JSON blob.
+        """
+        # need to do this deserialization round-trip to get us a dict
+        # that's serializable as JSON in the db:
+        # pydantic's `.dict()` is NOT serializable as JSON --
+        # see https://github.com/pydantic/pydantic/issues/1409
+        config_dict = loads(config.json())
+        return cls.create_or_update(db, data={"config_set": config_dict})
+
+    @classmethod
+    def get_resolved_config_property(
+        cls, db: Session, config_property: str, default_value: Any = None
+    ) -> Optional[Any]:
+        """
+        Gets the 'resolved' config property based on api-set and config-set configs.
+        `config_property` is a dot-separated path to the config property,
+        e.g. `notifications.notification_service_type`.
+
+        Api-set values get priority over config-set, in case of conflict.
+        """
+        config_record = db.query(cls).first()
+        if config_record:
+            api_prop = _get_property(config_record.api_set, config_property)
+            if api_prop is None:
+                logger.info(f"No API-set {config_property} property found")
+                return _get_property(
+                    config_record.config_set, config_property, default_value
+                )
+            return api_prop
+        logger.warning("No config record found!")
+        return default_value
+
+
+def _get_property(
+    properties: Any, property_path: str = None, default_value: Any = None
+) -> Any:
+    """
+    Internal helper to recursively traverse a property dict/value with a
+    dot-separated `property_path` string
+    """
+    if property_path:
+        if not isinstance(properties, dict):
+            raise ValueError("Invalid path specified")
```

Comment:
well the idea here is just that if we have both a non-empty/null `property_path` _and_ `properties` isn't a `dict`, then  something's gone wrong. but i think you're right that it's not necessarily the `property_path`, it could also just be a bogus `properties` object, if that was given to the first call of the function. so i'll adjust the error to make it a bit less specific.

really no code should be getting here, since we're not exposing this function directly. but definitely want to make it as robust as possible!